src/quicklist.c

```diff
@@ -1574,6 +1589,8 @@ int quicklistPopCustom(quicklist *quicklist, int where, unsigned char **data,
         return 1;
     }
 
+    assert(node->encoding != QUICKLIST_NODE_ENCODING_LZF);
```

Comment:
Maybe we should move it to before `if (unlikely(QL_NODE_IS_PLAIN(node)))`?
Plain node may also be incorrectly compressed.